File: BOJ/java/17087.java
/*
 * 숨바꼭질 6
 * https://www.acmicpc.net/problem/17087
 */

import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int s = sc.nextInt();
        int[] distances = new int[n];
        for (int i = 0; i < n; i++) {
            int b = sc.nextInt();
            distances[i] = Math.abs(s - b);
        }
        int result = distances[0];
        for (int i = 1; i < n; i++) {
            result = getGCD(result, distances[i]);
        }
        System.out.println(result);
        sc.close();
    }
    
    static int getGCD(int a, int b) {
        if (b == 0) {
            return a;
        } else {
            return getGCD(b, a % b);
        }
    }
}

File: BOJ/kotlin/17087.kt
import kotlin.math.abs

fun main() {
    val (n, s) = readLine()!!.split(" ")
    val distances = readLine()!!.split(" ")
        .map { it.toInt() }
        .map { s.toInt() - it }
        .map { abs(it) }
    var result = distances[0]
    (1 until n.toInt()).forEach {
        result = getGCD(result, distances[it])
    }
    println(result)
}

fun getGCD(a: Int, b: Int): Int = if (b == 0) a else getGCD(b, a % b)
